### ISSUE TITLE
Bug when setting catches to NA

### ISSUE THREAD

**Issue by Cole-Monnahan-NOAA** [James-Thorson-NOAA/VAST#246]
I noticed that VAST behaves inconsistently when you can set catches to NA. Sometimes it fails to setup the betas correctly and crashes. Here's a reproducible example below. This is for

````
> packageVersion('VAST')
[1] '3.4.0'
> packageVersion('FishStatsUtils')
[1] '2.7.0'
````

fit1 and fit3 behave as expected. fit2 crashes.

````
library(VAST)
dat <- load_example()$sampling_data

settings <- make_settings(n_x=500, Region='eastern_bering_sea',
                          fine_scale=FALSE, purpose='Index2',
                          bias.correct=FALSE, max_cells=Inf)
settings$FieldConfig[1:2,] <- 0

fit1 <- fit_model(settings, b_i=dat$Catch_KG,
                 Lat_i=dat$Lat, Lon_i=dat$Lon,
                 t_i=as.numeric(as.factor(dat$Year)),
                 a_i=dat$AreaSwept_km2)

## Set some random values to NA and it fails
dat2 <- dat
set.seed(342)
dat2$Catch_KG[sample(1:nrow(dat), size=100)] <- NA
fit2 <- fit_model(settings, b_i=dat2$Catch_KG,
                 Lat_i=dat2$Lat, Lon_i=dat2$Lon,
                 t_i=as.numeric(as.factor(dat2$Year)),
                 a_i=dat2$AreaSwept_km2)
#### The betas are not right (missing):
## List of estimated fixed and random effects:
##   Coefficient_name Number_of_coefficients  Type
## 1        logSigmaM                      1 Fixed

## But this works?
dat3 <- dat
dat3$Catch_KG[1:100] <- NA
fit3 <- fit_model(settings, b_i=dat3$Catch_KG,
                 Lat_i=dat3$Lat, Lon_i=dat3$Lon,
                 t_i=as.numeric(as.factor(dat3$Year)),
                 a_i=dat3$AreaSwept_km2)

````

**Comment by Cole-Monnahan-NOAA** [comment]
I get a new error when I do this, but it is probably unrelated. I had to set the Version to 9_2_0 manually or it threw an error with the default of 9_2_4. Ignoring that, this fix seems to work so I am closing this issue.

**Comment by James-Thorson** [comment]
Is it the compiler issue fixed in release 3.5.1? Or anything to fix in the
new issue?

**Comment by Cole-Monnahan-NOAA** [comment]
It's all good. I reinstalled VAST and FishStatsUtils clean and it's gone now. Don't understand why b/c those were the same versions I had but it is working. Fixed.
